server/deletion.go

```diff
@@ -32,79 +35,134 @@ func purgeUsers(db *sql.DB, pluginClient *pluginapi.Client, socketClient *model.
-	_, err = db.Exec(fmt.Sprintf(`
-DELETE FROM threadmemberships
-  WHERE postid IN (
-    SELECT id
-    FROM posts
-      WHERE posts.userid = '%s'
-  );
-`, userID))
-	if err != nil {
-		return fmt.Errorf("error when trying to delete user thread members: %s", err.Error())
-	}
+		query := sq.Select("Id").
+			From("Posts").
+			Where(sq.Eq{"UserId": userID}).
+			Limit(1000).
+			PlaceholderFormat(sq.Dollar)
 
-	// Delete reactions related to User's posts
-	_, err = db.Exec(fmt.Sprintf(`
-DELETE FROM reactions
-  WHERE postid IN (
-    SELECT id
-    FROM posts
-      WHERE posts.userid = '%s'
-  );
-`, userID))
-	if err != nil {
-		return fmt.Errorf("error when trying to delete user post reactions: %s", err.Error())
-	}
+		queryString, args, err := query.ToSql()
+		if err != nil {
+			return fmt.Errorf("error when trying to build the query: %s", err.Error())
+		}
 
-	// Delete replies to User's posts
-	_, err = db.Exec(fmt.Sprintf(`
-DELETE FROM posts
-  WHERE rootid IN (
-    SELECT id
-    FROM posts
-      WHERE posts.userid = '%s'
-  );
-`, userID))
-	if err != nil {
-		return fmt.Errorf("error when trying to delete replies to user posts: %s", err.Error())
-	}
+		rows, err := tx.Query(queryString, args...)
+		if err != nil {
+			return fmt.Errorf("error when trying to select user posts: %s", err.Error())
+		}
+		defer rows.Close()
 
-	// Delete User's posts
-	_, err = db.Exec(fmt.Sprintf(`
-DELETE FROM posts
-  WHERE posts.userid = '%s';
-`, userID))
-	if err != nil {
-		return fmt.Errorf("error when trying to delete user posts: %s", err.Error())
+		ids := []string{}
+		for rows.Next() {
+			var id string
+			if err = rows.Scan(&id); err != nil {
+				return fmt.Errorf("error when scanning row: %s", err.Error())
+			}
+			ids = append(ids, id)
+		}
+
+		if len(ids) == 0 {
+			break
+		}
+
+		deleteThreadsQuery := sq.Delete("Threads").
+			Where(sq.Eq{"postid": ids}).
+			PlaceholderFormat(sq.Dollar)
+
+		deleteThreadsQueryString, deleteThreadsArgs, err := deleteThreadsQuery.ToSql()
+		if err != nil {
+			return fmt.Errorf("error when trying to build the delete threads query: %s", err.Error())
+		}
+
+		_, err = tx.Exec(deleteThreadsQueryString, deleteThreadsArgs...)
+		if err != nil {
+			return fmt.Errorf("error when trying to delete user threads: %s", err.Error())
+		}
+
+		deleteThreadMembershipsQuery := sq.Delete("ThreadMemberships").
+			Where(sq.Eq{"postid": ids}).
+			PlaceholderFormat(sq.Dollar)
+
+		deleteThreadMembershipsQueryString, deleteThreadMembershipsArgs, err := deleteThreadMembershipsQuery.ToSql()
+		if err != nil {
+			return fmt.Errorf("error when trying to build the delete thread memberships query: %s", err.Error())
+		}
+
+		_, err = tx.Exec(deleteThreadMembershipsQueryString, deleteThreadMembershipsArgs...)
+		if err != nil {
+			return fmt.Errorf("error when trying to delete user thread memberships: %s", err.Error())
+		}
+
+		queryString, args, err = sq.Delete("Reactions").
```

Comment:
**nitpick:** we're using a unique variables for each query except this one. Should we create new `deleteReactionsQuery` and `deleteReactionsArgs` variables here?